SignFile can throw when the Windows SDK isn't installed

This callstack can occur on a Willow (new installer) machine that tries to sign a file:

```
error MSB4018: The "SignFile" task failed unexpectedly.\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018: System.ArgumentNullException: Value cannot be null.\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018: Parameter name: path1\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018:    at System.IO.Path.Combine(String path1, String path2, String path3)\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018:    at Microsoft.Build.Tasks.Deployment.ManifestUtilities.SecurityUtilities.GetPathToTool(ResourceManager resources)\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018:    at Microsoft.Build.Tasks.Deployment.ManifestUtilities.SecurityUtilities.SignPEFile(X509Certificate2 cert, Uri timestampUrl, String path, ResourceManager resources, Boolean useSha256)\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018:    at Microsoft.Build.Tasks.Deployment.ManifestUtilities.SecurityUtilities.SignFileInternal(X509Certificate2 cert, Uri timestampUrl, String path, Boolean targetFrameworkSupportsSha256, ResourceManager resources)\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018:    at Microsoft.Build.Tasks.Deployment.ManifestUtilities.SecurityUtilities.SignFile(X509Certificate2 cert, Uri timestampUrl, String path)\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018:    at Microsoft.Build.Tasks.Deployment.ManifestUtilities.SecurityUtilities.SignFile(String certThumbprint, Uri timestampUrl, String path, String targetFrameworkVersion)\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018:    at Microsoft.Build.Tasks.SignFile.Execute()\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()\r [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [c:\Users\raines\Documents\Visual Studio 15\Projects\WindowsFormsApplication1\WindowsFormsApplication1\WindowsFormsApplication1.csproj]
```

That's coming from https://github.com/Microsoft/msbuild/blob/7ed1c641059428cfca1cd91887fdc8aed289e26d/src/XMakeTasks/ManifestUtil/SecurityUtil.cs#L768,

```C#
toolPath = Path.Combine(ToolLocationHelper.GetPathToDotNetFrameworkSdk(TargetDotNetFrameworkVersion.Version40, VisualStudioVersion.Version100), "bin", ToolName);
```

In a scenario where the .NET SDK 4.0 isn't installed, `GetPathToDotNetFrameworkSdk` returns `null`, causing `Path.Combine` to throw.

We shouldn't throw in that case.

--- a/src/XMakeTasks/ManifestUtil/SecurityUtil.cs
+++ b/src/XMakeTasks/ManifestUtil/SecurityUtil.cs
@@ -765,7 +765,13 @@ namespace Microsoft.Build.Tasks.Deployment.ManifestUtilities
             if (toolPath == null)
                 toolPath = ToolLocationHelper.GetPathToWindowsSdkFile(ToolName, TargetDotNetFrameworkVersion.Version45, VisualStudioVersion.Version110);
             if (toolPath == null)
-                toolPath = Path.Combine(ToolLocationHelper.GetPathToDotNetFrameworkSdk(TargetDotNetFrameworkVersion.Version40, VisualStudioVersion.Version100), "bin", ToolName);
+            {
+                var pathToDotNetFrameworkSdk = ToolLocationHelper.GetPathToDotNetFrameworkSdk(TargetDotNetFrameworkVersion.Version40, VisualStudioVersion.Version100);
+                if (pathToDotNetFrameworkSdk != null)
+                {
+                    toolPath = Path.Combine(pathToDotNetFrameworkSdk, "bin", ToolName);
+                }
+            }
             if (toolPath == null)
                 toolPath = Path.Combine(Directory.GetCurrentDirectory(), ToolName);
             if (!File.Exists(toolPath))

